File: data_types_lab.R
########## Q1

questionOne <- c("character", "logical", "integer", "double")

########## Q2

a <- 2 / 3
print(a)
aPrint <- 0.6666667

numDigitsStart <- options()$digits

options(digits = 22)

print(a)

aPrintv2 <- 0.6666666666666666296592

numDigitsEnd <- options()$digits

options(digits = numDigitsStart)

########## Q3

mySqrt <- sqrt(2)

print(mySqrt**2)

isTwoRecoverable <- (2 == (mySqrt**2))

closeEnough <- !is.character(all.equal(mySqrt^2, 2))

########## Q4

flips <- rbinom(n = 1000, size = 1, prob = 0.5)

isHeads <- flips == 1

numHeads <- sum(isHeads)

acceptanceRate <- (sum(flips[-1] != flips[-length(flips)]) / (length(flips) - 1)) * 100

File: data_types_lab.py
import math
import numpy as np

########## Q1

question_one = ["str", "bool", "int", "float"]

########## Q2

a = 2 / 3

print(a)

a_print = '0.6666666666666666'

a_printv2 = f"{a:.22f}"

print(a_printv2)

########## Q3

my_sqrt = math.sqrt(2)

print(my_sqrt**2)

is_two_recoverable = (my_sqrt**2 == 2)

close_enough = np.isclose(my_sqrt**2, 2)

########## Q4

flips = np.random.binomial(size = 1000, n = 1, p = 0.5)

is_heads = (flips == 1).tolist()

num_heads = np.sum(flips)

acceptance_rate = np.mean(flips[:-1] != flips[1:]) * 100